### PR TITLE
Updated Readme

### PR DESCRIPTION
**Pull request by SameerK-MSFT**
**Changes proposed in this request**
Updated readme for the release

**Testing**
NA

**Performance impact**
NA

### CODE REVIEW

**Review by jennyf19 — CHANGES_REQUESTED** [review]
see comment in teams.

**Review by jmprieur — CHANGES_REQUESTED** [review]
This is not as customer focused as the previous changes for the previous versions, @SameerK-MSFT

File: changelog.txt [hunk 1]
```diff
@@ -1,3 +1,41 @@
+4.35.0
+==========
+### Bug Fixes:
+**AcquireTokenByIntegratedWindowsAuth against ADFS provides clearer error message** Error messages have been udpated. See issue [#2752](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2752)
```
Comment (trwalke) [inline changelog.txt:4]:
```suggestion
**AcquireTokenByIntegratedWindowsAuth provides better error messages.** Error messages are now more actionable. See issue [#2752](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2752) 
```

File: changelog.txt [hunk 2]
```diff
@@ -1,3 +1,41 @@
...
+==========
+### Bug Fixes:
+**AcquireTokenByIntegratedWindowsAuth against ADFS provides clearer error message** Error messages have been udpated. See issue [#2752](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2752) 
+
+**MSAL no longer performs instance discovery on well known authorities.** This will improve performance for customers in regional scenarios See issue  [#2777](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2777)
+
+**MSAL uses preferred_network name on sovereign clouds.** Skipping discovery will improve the performance. See issue [#2778](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2778)
+
+
+**Error messages in Integrated Windows Authentication are now clearer**. The following message related issues are fixed
```
Comment (trwalke) [inline changelog.txt:11]: Extra line

File: changelog.txt [hunk 3]
```diff
@@ -1,3 +1,41 @@
...
+    - [#2731](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2731)
+    - [#2752](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2752)
+
+**ConfidentialClientApplicationBuilder with auto region discovery no longer throws UriFormatException.** This has been fixed by validating region string. See issue [#2772](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2772)
+
+**Memory leak in AuthorityEndpoint caching has been fixed.** This will reduce memory leaks in the apps. See issue [#2770](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2770)
+
+**ADFS now has consistent values for UserName between STS and cache.**. Tenant profiles will provide the consistency. [#1559](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1559)
+
+**MSAL.NET would throw an ArgumentNullException if the provided CcsRoutingHint was null**, now authentication will proceed forward as providing the hint is for resiliency and should not prevent sign-in. See issue [#2755](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2755)
```
Comment (trwalke) [inline changelog.txt:21]:
```suggestion
**MSAL.NET no longer throws ArgumentNullException when the parameters for WithCcsRoutingHint() are null.** Authentication will now proceed as if not hint was provided. See issue [#2755](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2755)
```

File: changelog.txt [hunk 4]
```diff
@@ -1,3 +1,41 @@
...
+
+**The mismtached authorities were throwing exceptions.** MSAL now does instance discovery to find the right match. See issue [#2736](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2736)
+
+**WebView2 was throwing error when KeyDown was handled.** This error has been removed. See issue [#2685](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2685)
+
+**Instance discovery was being performed multiple times on non-public non-sovereign clouds.** You will get better performance as now it does discovery only if it is not in the cache. See issue [#2701](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2701)
```
Comment (trwalke) [inline changelog.txt:27]:
```suggestion
**Instance discovery was performed multiple times on non-public non-sovereign clouds.** MSAL provides improved performance as now it does discovery only when needed. See issue [#2701](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2701)
```

File: changelog.txt [hunk 5]
```diff
@@ -1,3 +1,41 @@
...
+
+**Memory leak in AuthorityEndpoint caching has been fixed.** This will reduce memory leaks in the apps. See issue [#2770](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2770)
+
+**ADFS now has consistent values for UserName between STS and cache.**. Tenant profiles will provide the consistency. [#1559](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1559)
+
+**MSAL.NET would throw an ArgumentNullException if the provided CcsRoutingHint was null**, now authentication will proceed forward as providing the hint is for resiliency and should not prevent sign-in. See issue [#2755](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2755)
+
+**The mismtached authorities were throwing exceptions.** MSAL now does instance discovery to find the right match. See issue [#2736](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2736)
```
Comment (trwalke) [inline changelog.txt:23]:
```suggestion
**MSAL no longer throws exceptions for mismatched authorities if they are known aliases.** MSAL now ensures authorities configured in the application and request are not aliased before throwing. See issue [#2736](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2736)
```

File: changelog.txt (hunk 2 above)
Comment (SameerK-MSFT) [reply changelog.txt:10]: Removed

**Review by trwalke — APPROVED** [review]
Thanks Sameer

**Comment by SameerK-MSFT** [comment]
Readme has been updated. Let me know, if any further changes are required.

**Pull request merged**